catalog/video/video-security/src/test/java/org/codice/alliance/video/security/claims/videographer/VideographerClaimsHandlerTest.java

```diff
@@ -80,94 +93,195 @@ public void testSettingClaimsMapList() throws URISyntaxException {
+        claimsCollection = claimsHandler.retrieveClaimValues(requestClaims, claimsParameters);
 
-        assertThat(supportedClaims, hasSize(3));
+        assertThat(claimsCollection, contains(getIpClaim()));
+
+    }
+
+    @Test
+    public void testRetrieveClaimsForVideographerPrincipal() throws URISyntaxException {
+
+        createClaimsHandler();
 
-        ProcessedClaimCollection claimsCollection = claimsHandler.retrieveClaimValues(requestClaims,
-                claimsParameters);
+        createClaims();
 
-        Matcher<ProcessedClaim> nameClaim = new CustomMatcher<ProcessedClaim>("claim by name") {
+        claimsParameters = new ClaimsParameters();
+
+        claimsParameters.setPrincipal(mock(VideographerPrincipal.class));
+        claimsCollection = claimsHandler.retrieveClaimValues(requestClaims, claimsParameters);
+
+        assertThat(claimsCollection, hasSize(2));
+
+    }
+
+    @Test
+    public void testRetrieveClaimsForNonVideographerPrincipal() throws URISyntaxException {
+
+        createClaimsHandler();
+
+        createClaims();
+
+        claimsParameters = new ClaimsParameters();
+
+        claimsParameters.setPrincipal(new CustomTokenPrincipal("SomeValue"));
+        claimsCollection = claimsHandler.retrieveClaimValues(requestClaims, claimsParameters);
+
+        assertThat(claimsCollection, hasSize(0));
+    }
+
+    private Matcher<ProcessedClaim> getNameClaim(URI uri) {
+
+        return new CustomMatcher<ProcessedClaim>("claim by name") {
             @Override
             public boolean matches(Object o) {
                 ProcessedClaim claim = (ProcessedClaim) o;
                 return claim.getClaimType()
-                        .equals(nameURI) && claim.getValues()
+                        .equals(uri) && claim.getValues()
                         .size() == 1 && claim.getValues()
                         .get(0)
                         .equals(CLAIM_VALUE_1);
             }
         };
+    }
 
-        Matcher<ProcessedClaim> emailClaim = new CustomMatcher<ProcessedClaim>("claim by email") {
+    private Matcher<ProcessedClaim> getEmailClaim(URI uri) {
+
+        return new CustomMatcher<ProcessedClaim>("claim by email") {
             @Override
             public boolean matches(Object o) {
                 ProcessedClaim claim = (ProcessedClaim) o;
                 return claim.getClaimType()
-                        .equals(emailURI) && claim.getValues()
-                        .size() == 3 &&
-                        claim.getValues()
-                                .get(0)
-                                .equals(CLAIM_VALUE_2A) &&
-                        claim.getValues()
-                                .get(1)
-                                .equals(CLAIM_VALUE_2B) &&
-                        claim.getValues()
-                                .get(2)
-                                .equals(CLAIM_VALUE_2C);
+                        .equals(uri) && claim.getValues()
```

Comment:
You _could_ replace the size and value checks with `IsIterableContainingInOrder.contains(CLAIM_VALUE_2A, CLAIM_VALUE2B, CLAIM_VALUE2C).matches(claim.getValues())`.